cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
```

Comment:
Yes, except that the variable argument list can't contain that specific keyword, so I think `context` would be "safe". It needs to be extracted and used with something like this:

```
cmake_parse_arguments(ARG "" "CONTEXT" "" ${ARGN})
# ARG_CONTEXT
# ARG_UNPARSED_ARGUMENTS
```